webrender/src/display_list_flattener.rs

```diff
@@ -1029,6 +1029,21 @@ impl<'a> DisplayListFlattener<'a> {
             None
         };
 
+        // preserve-3d's semantics are to hoist all your children to be your siblings
+        // when doing backface-visibility checking, so we need to grab the backface-visibility
+        // of the lowest ancestor which *doesn't* preserve-3d, and AND it in with ours.
```

Comment:
I think the proper thing to do would be to grab the closest ancestor that is still `Preserve3D` but *whose parent* is `Flat`